### PR TITLE
Add Snippet Toggler to Docs

### PR DESCRIPTION
**Pull request by estheragbaje**
This PR adds a snippet toggler to the hero section of the docs landing page

### CODE REVIEW

File: docs/.vitepress/components/SnippetToggler.vue [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+<template>
+	<div class="snippet-toggler">
+		<div class="snippet-toggler-header">
+			<span class="snippet-toggler-header-label">{{ label }}</span>
+
+			<span class="spacer" />
+
+			<div class="snippet-toggler-header-lang-container">
+				<select class="snippet-toggler-header-lang" :value="pref" @change="onPrefChange">
```
Comment (rijkvanzanten) [inline docs/.vitepress/components/SnippetToggler.vue:25]: Vue has a convenient shorthand for this type of two-way binding 🙂 

```suggestion
				<select class="snippet-toggler-header-lang" v-model="pref">
```

File: docs/.vitepress/components/SnippetToggler.vue [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+</template>
+
+<style scoped>
+.snippet-toggler {
+	overflow: hidden;
+	background: var(--vp-code-block-bg);
+}
+.snippet-toggler-header {
+	background-color: #25232d;
+	color: #a6accd;
```
Comment (rijkvanzanten) [inline docs/.vitepress/components/SnippetToggler.vue:59]: Lets be careful with hardcoded hex values (re theming; dark/light mode)

File: docs/.vitepress/components/SnippetToggler.vue [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+.content-area {
+	padding-inline: 24px;
+	padding-bottom: 32px;
+	scrollbar-width: none;
+	overflow-y: scroll;
+}
+
+.content-area::-webkit-scrollbar {
+	display: none;
+}
```
Comment (rijkvanzanten) [inline docs/.vitepress/components/SnippetToggler.vue:116]: Not entirely sure what the goal is, but force-hiding the scrollbar here feels tricky. Are we trying to make it overflowing without a visible scrollbar? Can we use `overflow-y: auto` instead so it only shows a scrollbar when the content overflows?

File: docs/index.md [hunk 4]
```diff
@@ -25,7 +25,77 @@ layout: home
         >
       </div>
     </div>
-    <CodeToggler class="hero-toggler" />
+    <div class="hero-toggler">
+
+<SnippetToggler initial="REST" :choices="['REST', 'GraphQL', 'JS-SDK']" label="API">
```
Comment (rijkvanzanten) [inline docs/index.md:30]: Is there ever a case where the first item in `choices` shouldn't be the `initial`?

Comment (rijkvanzanten) [reply docs/index.md:30]: & Lets just call it `SDK` instead of `JS-SDK`

File: docs/.vitepress/components/SnippetToggler.vue (hunk 1 above)
Comment (estheragbaje) [reply docs/.vitepress/components/SnippetToggler.vue:25]: I tried this and it didn't work for some reason so used this method

File: docs/.vitepress/theme/overrides.css [hunk 5]
```diff
@@ -53,3 +53,19 @@
 	max-width: calc(var(--vp-layout-max-width) - 64px);
 	margin: 0 auto;
 }
+
+/**
+ * Snippet Toggler Content
```
[the remark this answers is not in the document]
Comment (estheragbaje) [reply docs/.vitepress/theme/overrides.css:58]: yes, moved it

**Pull request merged**